python/ModelTools.py

```diff
@@ -808,7 +810,7 @@ def doFillNuisPdfsAndSets(self):
                 if p != "constr":
                     nuisVars.add(self.out.var(c_param_name))
                 setNuisPdf.append(c_param_name)
-            setNuisPdf = set(setNuisPdf)
+            setNuisPdf = list(dict.fromkeys((setNuisPdf)))
```

Comment:
I did a few tests, and it seems that this line (and line 826 below) changing the sets to an ordered type are enough for exact reproducibility of the fit values on the datacards where we saw the issue when recreating the workspace. Though it may be worth understanding this more clearly. 

-- On the broader point I agree with  your suggestion, @nsmith- it would be nice to make those kinds of changes.